scripts/delete-old-app-engine-versions.sh

```diff
@@ -0,0 +1,184 @@
+
+    UNIX_EPOCH_DELETION_DATE=$(date --date="${DELETION_DATE}" +%s)
+    UNIX_EPOCH_OLDEST_PR_DATE="$(date --date="${OLDEST_PR_DATE}" +%s)"
+
+    if [ "${UNIX_EPOCH_DELETION_DATE}" -gt "${UNIX_EPOCH_OLDEST_PR_DATE}" ]; then
+        DELETION_DATE=$(date --date="${OLDEST_PR_DATE} -1 day" +%F)
+    fi
+}
+
+# return versions of app engine that match filter
+filter_app_engine_versions() {
+    gcloud app versions list --filter="$1" 2>/dev/null | tail -n +2 | sed 's/ \+/ /g' | cut -d' ' -f2
+}
+
+# ensure that deletions leave a certain number of deployments
+check_remaining_items() {
+    REMAIN_LIST_ITEMS=($(filter_app_engine_versions "version.createTime.date('%Y-%m-%d', Z)>'${DELETION_DATE}'"))
+    REMAIN_LIST_COUNT="${#REMAIN_LIST_ITEMS[@]}"
+    if [ "${REMAIN_LIST_COUNT}" -lt 1 ]; then
+        abort "all deployments would be deleted"
+    elif [ "$REMAIN_LIST_COUNT" -lt 15 ]; then
+        abort "less than 15 deployments would remain"
+    fi
+}
+
+# ensure that deletions erase a certain number of deployments
+check_deletion_items() {
+    DELETE_LIST_ITEMS=($(filter_app_engine_versions "version.createTime.date('%Y-%m-%d', Z)<='${DELETION_DATE}'"))
+    DELETE_LIST_COUNT="${#DELETE_LIST_ITEMS[@]}"
+    if [ "${DELETE_LIST_COUNT}" -lt 1 ]; then
+        abort "no deployments to delete"
+    elif [ "${DELETE_LIST_COUNT}" -lt 30 ]; then
+        abort "less than 30 deployments to delete"
+    fi
+}
+
+# run preflight checks to minimize deletion errors
+deletion_preflight_checks() {
+    check_remaining_items
+    check_deletion_items
+}
+
+# print out a summary of the deployments that will be affected
+deletion_preflight_summary() {
+    printf "${INFO} This script will ${GRN}keep %d${RST} deployments and ${RED}delete %d${RST} deployments\n" "${REMAIN_LIST_COUNT}" "${DELETE_LIST_COUNT}"
+    printf "${INFO} Deployments to keep (check this list):\n[ %s ]\n\n" "${REMAIN_LIST_ITEMS[*]}"
+}
+
+# actually execute the deletion process
+execute_delete() {
+    printf "${BLD}gcloud app versions delete %s${RST}\n" "${DELETE_LIST_ITEMS[*]}"
+}
+
+# ensure that a deletion phrase must be entered correctly before continuing
+enter_deletion_phrase() {
+    DELETION_PHRASE="yes delete ${DELETE_LIST_COUNT} deployments in ${ENV_TO_EXEC}"
+    printf "${RED}THIS OPERATION WILL IRREVERSIBLY DELETE ${DELETE_LIST_COUNT} DEPLOYMENTS FROM %s AND EARLIER IN THE ${NEW_PROJECT} PROJECT.${RST}\n" "${DELETION_DATE}"
+    printf "${INFO} To continue with the deletion process, type: ${BLD}%s${RST}\n" "${DELETION_PHRASE}"
+    read -r ACTUAL_PHRASE
+    if [ "${ACTUAL_PHRASE}" = "${DELETION_PHRASE}" ]; then
+        execute_delete
+    else
+        abort "mistyped phrase"
+    fi
+}
+
+check_color_support
+
+check_gnu_date_installed
+
+if [ -z "${1+:}" ]; then
+    usage
+fi
+
+case $1 in
+    --help ) usage;;
+    dev ) ;;
+    alpha ) ;;
+    staging ) ;;
+    perf ) ;;
```

Comment:
You can combine cases with the same body
```suggestion
    dev|alpha|staging|perf) ;;
```